### DISCUSSION TITLE
P2G and Garmin cycling training plans

### DISCUSSION THREAD

**Discussion by velozriding** [philosowaffle/peloton-to-garmin#560] [Q&A] [answered]
Hi there, I have a century ride coming up in a few months and want to try out one of the Garmin Connect cycling century training plans. I haven’t used one before, just trained myself. Ideally I want to do some of the rides on the Peloton bike and push the data to Garmin through P2G and do the remainder of the rides outside as my schedule allows. 

Has anyone done this before and know if the Garmin Connect training plan will recognize the Peloton rides as part of the training plan? Thanks in advance for any help/insights.

**Comment by philosowaffle** [comment]
I have not personally tried this.  Generally I can't see any reason why these Peloton workouts wouldn't count towards the training plan other than perhaps stacked rides.  For instance if the training plan calls for a 3hr ride, and you choose to do that by stacking three 1hr Peloton classes, I'm not sure how the Training plan would handle counting that. I know that Garmin won't count such a thing towards Badges (mentioned in this [feature request](https://github.com/philosowaffle/peloton-to-garmin/issues/556)).

Its probably worth experimenting with in advance with a throw away Training Plan on Garmin just so you can have confidence in what to do for the real plan.

**Reply by velozriding** [reply, marked as answer]
Thanks so much for the response. I’ve been experimenting and it seems you can only “complete” a workout from a Garmin plan if you are doing it directly on a Garmin device like a bike computer or a watch; on the device you choose to “start” that particular workout that is part of your training plan. I’m still trying to just mimic the workout on the Peloton bike (either in scenic or “just ride” mode) to follow the plan for training purposes even it Garmin won’t recognize as such. At the end of the day, I just need legs ready for 100 miles! :-)

**Reply by jayflan** [reply]
Hi.  Same here.  Can't "complete" a training plan ride if not recording 100% from watch/device.  Though I do appreciate the workout guidance from the Watch to get the job done. 

Wonder how folks using Tacx trainers get it done?  If there's a part of the data that helps evaluate or notify the plan that a workout was completed.
